src/main/java/com/ldtteam/structurize/util/IOPool.java

```diff
@@ -0,0 +1,92 @@
+package com.ldtteam.structurize.util;
+
+import com.ldtteam.structurize.api.util.Log;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.concurrent.*;
+
+/**
+ * Class handling our IO pool.
+ */
+public final class IOPool
+{
+    /**
+     * Hold our IO queue.
+     */
+    private static final BlockingQueue<Runnable> ioQueue = new LinkedBlockingDeque<>();
+
+    /**
+     * Holds the specific executor for the queue.
+     */
+    private static ThreadPoolExecutor executor;
+
+    /**
+     * Structurize specific thread factory.
+     */
+    public static class StructurizeThreadFactory implements ThreadFactory
+    {
+        /**
+         * Ongoing thread IDs.
+         */
+        public static int id;
```

Comment:
while the executor is static this id being static is very sus, either non-static or atomic